pkg/network/node/networkpolicy.go

```diff
@@ -500,6 +500,8 @@ func (np *networkPolicyPlugin) parseNetworkPolicy(npns *npNamespace, policy *net
 		// that affects ingress but does not select any pods.
 		npp.selectedIPs = nil
 		npp.selectsAllIPs = true
+		// add an empty flow to allow pods to talk when there is an empty Egress policy
+		npp.flows = append(npp.flows, fmt.Sprintf("%s", ""))
```

Comment:
I think the bug here is that it's setting `npp.selectsAllIPs`; an egress-only policy is supposed to behave like it selects *no* IPs (like the comment above says). So I think the fix is just to change to `npp.selectsAllIPs = false`. You don't need to create an empty flow.

Reply:
OK, thanks for the review. I have made the changes as you suggested. Will spin up a cluster and test to verify.